clang/test/CheckedCRewriter/arrboundsmerging.c

```diff
@@ -0,0 +1,55 @@
+// RUN: cconv-standalone -disable-arr-hu -alltypes %s -- | FileCheck -match-full-lines %s
+
+
+/*
+Advanced array-bounds inference (based on control-dependencies).
+*/
+
+typedef unsigned long size_t;
+extern _Itype_for_any(T) void *malloc(size_t size) : itype(_Array_ptr<T>) byte_count(size);
+
+// Here x bounds will be c
+void foo(int *x, int c) {
+//CHECK: void foo(_Array_ptr<int> x : count(c), int c) {
```

Comment:
Seems odd that this should be `c` since the index of `3` is not at all related to `c`. I.e., the index is unconditional.

Reply:
Maybe this is because of the heuristic (which we are not disabling) ?

Reply:
This is not a heuristic. It is from all the incoming array pointers i.e., `p`, `q`. i.e., all the calls to this function pass array to `x` and bounds to `c`. We do not worry about the indexing `x` as Checked C dynamic checks will take care of that.